spec/tapioca/compilers/dsl/active_record_typed_store_spec.rb

```diff
@@ -0,0 +1,260 @@
+          def reviewed_changed?; end
+
+          sig { returns(T::Boolean) }
+          def reviewed_was; end
+
+          sig { returns(T.nilable([T::Boolean, T::Boolean])) }
+          def saved_change_to_reviewed; end
+
+          sig { returns(T::Boolean) }
+          def saved_change_to_reviewed?; end
+        end
+        RUBY
+
+      expect(rbi_for(content)).to(eq(expected))
+    end
+
+    it("generate RBI for simple TypedStore classes with date type ") do
+      content = <<~RUBY
+        class Post < ActiveRecord::Base
+          typed_store :metadata do |s|
+            s.date(:review_date)
+          end
+        end
+      RUBY
+
+      expected = <<~RUBY
+        # typed: strong
+        class Post
+          sig { returns(T.nilable(Date)) }
+          def review_date; end
+
+          sig { params(review_date: T.nilable(Date)).returns(T.nilable(Date)) }
+          def review_date=(review_date); end
+
+          sig { returns(T::Boolean) }
+          def review_date?; end
+
+          sig { returns(T.nilable(Date)) }
+          def review_date_before_last_save; end
+
+          sig { returns(T.nilable([T.nilable(Date), T.nilable(Date)])) }
+          def review_date_change; end
+
+          sig { returns(T::Boolean) }
+          def review_date_changed?; end
+
+          sig { returns(T.nilable(Date)) }
+          def review_date_was; end
+
+          sig { returns(T.nilable([T.nilable(Date), T.nilable(Date)])) }
+          def saved_change_to_review_date; end
+
+          sig { returns(T::Boolean) }
+          def saved_change_to_review_date?; end
+        end
+        RUBY
+
+      expect(rbi_for(content)).to(eq(expected))
+    end
+
+    it("generate RBI for simple TypedStore classes with datetime ") do
+      content = <<~RUBY
+        class Post < ActiveRecord::Base
+          typed_store :metadata do |s|
+            s.datetime(:review_date)
+          end
+        end
+      RUBY
+
+      expected = <<~RUBY
+        # typed: strong
+        class Post
+          sig { returns(T.nilable(DateTime)) }
+          def review_date; end
+
+          sig { params(review_date: T.nilable(DateTime)).returns(T.nilable(DateTime)) }
+          def review_date=(review_date); end
+        RUBY
+      expect(rbi_for(content)).to(include(expected))
+    end
+
+    it("generate RBI for simple TypedStore classes with decimal type ") do
+      content = <<~RUBY
+        class Post < ActiveRecord::Base
+          typed_store :metadata do |s|
+            s.decimal(:rate)
+          end
+        end
+      RUBY
+
+      expected = <<~RUBY
+        # typed: strong
+        class Post
+          sig { returns(T.nilable(BigDecimal)) }
+          def rate; end
+
+          sig { params(rate: T.nilable(BigDecimal)).returns(T.nilable(BigDecimal)) }
+          def rate=(rate); end
+        RUBY
+      expect(rbi_for(content)).to(include(expected))
+    end
+
+    it("generate RBI for simple TypedStore classes with any type ") do
```

Comment:
I don't see any tests for `Integer`, `Float` and maybe others. Can we be exhaustive please?

Reply:
I added those two, I think now, it should cover all these types:
          ` boolean: "T::Boolean",
            integer: "Integer",
            string: "String",
            float: "Float",
            date: "Date",
            time: "Time",
            datetime: "DateTime",
            decimal: "BigDecimal",
            any: "T.untyped"`

Reply:
I am not seeing any `Time` cases, I think that is the only one that is missing now.